dpdk/vr_dpdk_interface.c

```diff
@@ -801,13 +808,23 @@ dpdk_if_tx(struct vr_interface *vif, struct vr_packet *pkt)
 #endif
 
     if (likely(tx_queue->txq_ops.f_tx != NULL)) {
-        tx_queue->txq_ops.f_tx(tx_queue->q_queue_h, m);
-        if (lcore_id == vr_dpdk.packet_lcore_id)
-            tx_queue->txq_ops.f_flush(tx_queue->q_queue_h);
+        ret = tx_queue->txq_ops.f_tx(tx_queue->q_queue_h, m);
+        if (ret) {
+            vif_drop_pkt(vif, vr_dpdk_mbuf_to_pkt(m), 0, VP_DROP_ENQUEUE_FAIL);
+            return -1;
+        }
+
+        if (lcore_id == vr_dpdk.packet_lcore_id) {
+            ret = tx_queue->txq_ops.f_flush(tx_queue->q_queue_h);
+            if (ret) {
+                vif_drop_pkt(vif, vr_dpdk_mbuf_to_pkt(m), 0, VP_DROP_ENQUEUE_FAIL);
```

Comment:
When will the flush fail? Also, how do we know the number of packets that failed?